mos_tests/neutron/python_tests/test_ovs_restart.py

```diff
@@ -895,3 +906,126 @@ def test_ovs_new_flows_added_after_restart(self):
+        for i, hostname in enumerate(hosts, 5):
+            net, subnet = self.create_internal_network_with_subnet(suffix=i)
+            if i == 5:
+                self.os_conn.router_interface_add(
+                    router_id=router_05['router']['id'],
+                    subnet_id=subnet['subnet']['id'])
+            else:
+                self.os_conn.router_interface_add(
+                    router_id=router_06['router']['id'],
+                    subnet_id=subnet['subnet']['id'])
+            self.os_conn.create_server(
+                name='test_vm_%02d' % i,  # 'test_vm_05' OR 'test_vm_06'
+                availability_zone='{}:{}'.format(zone.zoneName, hostname),
+                key_name=self.instance_keypair.name,
+                timeout=200,
+                nics=[{'net-id': net['network']['id']}])
+
+        # Check pings with alive ovs-agents,
+        # and before restart 'neutron-plugin-openvswitch-agent'
+        self.server1 = self.os_conn.nova.servers.find(name="test_vm_06")
+        self.server2_ip = self.os_conn.get_nova_instance_ips(
+            self.os_conn.nova.servers.find(name="test_vm_05")
+        ).values()[0]
+
+        # Ping should NOT go between VMs
+        self.check_ping_from_vm(self.server1, self.instance_keypair,
+                                self.server2_ip, timeout=None,
+                                should_be_available=False)
+
+    def test_ovs_restart_pcs_disable_enable_ping_private_vms(self):
+        """Restart openvswitch-agents with pcs disable/enable on controllers
+        [VLAN only] Check connectivity between private networks on different
+        routers
+
+        Steps:
+        1. Update default security group if needed
+        2. Create CONFIG 1:
+        Network: test_net_05
+        SubNetw: test_net_05__subnet, 192.168.5.0/24
+        Router:  test_router_05
+        3. Create CONFIG 2:
+        Network: test_net_06
+        SubNetw: test_net_06__subnet, 192.168.6.0/24
+        Router:  test_router_06
+        3. Launch 'test_vm_05' inside 'config 1'
+        4. Launch 'test_vm_06' inside 'config 2'
+        5. Go to 'test_vm_05' console and send pings to 'test_vm_05'.
+        Pings should NOT go between VMs.
+        5. Disable ovs-agents on all controllers, restart service
+        neutron-plugin-openvswitch-agent on all computes, and enable
+        them back. To do this, launch the script against master node.
+        6. Wait 30 seconds, send pings from 'test_vm_05' to 'test_vm_06'
+        and check that they are still NOT successful.
+
+        Duration 10m
+
+        https://mirantis.testrail.com/index.php?/cases/view/542666
+        https://mirantis.jira.com/browse/QA-375
+        """
+        self.disable_ovs_agents_on_controller()
+        self.restart_ovs_agents_on_computes()
+        self.enable_ovs_agents_on_controllers()
+
+        # sleep is used to check that system will be stable for some time
+        # after restarting service
+        time.sleep(30)
```

Comment:
I'll add additional checks here, but anyway - put 30sec sleep was a recommendation from MOS team to have more stable env after agents restart